tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs

```diff
@@ -199,8 +200,8 @@ private void FindPattern(string message)
         {
             _dispatcher.TryEnqueue(() =>
             {
-                var insightsPageViewModel = Application.Current.GetService<InsightsPageViewModel>();
-                insightsPageViewModel.AddInsight(newInsight);
+                var insightsService = Application.Current.GetService<PIInsightsService>();
```

Comment:
WinLogsPageViewModel is a registered service, and so is PIInsightsService, so you can get the PIInsightsService in the constructor of this viewmodel. Dependency injection here would be useful if you ever are going to use this class in tests.